simplified-viewer-audiobook/src/main/java/org/nypl/simplified/viewer/audiobook/AudioBookPlayerActivity.kt

```diff
@@ -148,7 +148,7 @@ class AudioBookPlayerActivity : AppCompatActivity(),
     this.log.debug("entry id:      {}", this.parameters.opdsEntry.id)
 
     this.setTheme(
-      Services.serviceDirectory()
+      Services.serviceDirectoryWaiting(30L, TimeUnit.SECONDS)
```

Comment:
How is it that we're reaching this code?

Is this happening because Android restarts the app with the `AudioBookPlayerActivity` at the top of the stack, and the boot process is therefore happening in the background?

Reply:
I have no idea. The Application instance should be the first thing that starts. If services are initialized asynchronously it's possible there could be a race condition here if `AudioBookPlayerActivity` is at the top of the stack and the app is restarting. Perhaps if a user left the app open or backgrounded and came back a while later. The system could have killed the app process due to memory pressure and is restarting. Worth investigating more but I don't have time right now 😂